src/lib/xmlparse.rs

```diff
@@ -3115,15 +3101,17 @@ pub unsafe extern "C" fn XML_GetParsingStatus(
     }
     *status = (*parser).m_parsingStatus;
 }
-/* If XML_Parse or XML_ParseBuffer have returned XML_STATUS_ERROR, then
+/* If XML_Parse or XML_ParseBuffer have returned XML_Status::ERROR, then
    XML_GetErrorCode returns information about the error.
 */
 #[no_mangle]
-pub unsafe extern "C" fn XML_GetErrorCode(mut parser: XML_Parser) -> XML_Error {
-    if parser.is_null() {
-        return XML_ERROR_INVALID_ARGUMENT;
-    }
-    return (*parser).m_errorCode;
+pub unsafe extern "C" fn XML_GetErrorCode(mut parser: XML_Parser) -> XML_ErrorCode {
```

Comment:
Shouldn't `XML_ErrorCode` retain its `libc::c_uint` type rather than `u32`? Since it gets used in the C API as a return type?

Reply:
Changed in d909c1b. I left the `XML_Error` enum as `#[repr(u32)]` since the `repr` attribute does not accept a type alias and it seems that the `libc` crate always defines `c_int` and `c_uint` as `i32` and `u32` respectively.